vocab-importer/src/main.py

```
@@ -0,0 +1,286 @@
+                    raise VocabImporterError("No data received from backend")
+            except VocabImporterError as e:
+                raise e
+            except Exception as e:
+                raise VocabImporterError(
+                    f"Error fetching from backend: {str(e)}",
+                    status_code=status.HTTP_502_BAD_GATEWAY,
+                )
+
+            # Create temporary file
+            try:
+                temp_file = tempfile.NamedTemporaryFile(
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)